### PR TITLE
Check issue remains fixed

### PR DESCRIPTION
**Pull request by idontgetoutmuch**
Checks that https://github.com/haskell/random/issues/53 remains fixed

### CODE REVIEW

File: test/Spec.hs [hunk 1]
```diff
@@ -71,8 +72,21 @@ main =
...
     , runSpec
+    , unitTests
     ]
 
+unitTests :: TestTree
+unitTests = testGroup "Unit tests"
+  [ -- Check that https://github.com/haskell/random/issues/53 does not regress
+
+    testCase "Subnormal generation respects upper bound" $
+    [] @?= (filter (>4.0e-45) $ take 10 $ randomRs (0,(4.0e-45::Float)) $ mkStdGen 0)
```
Comment (curiousleo) [inline test/Spec.hs]:
```suggestion
    [] @?= (filter (>4.0e-45) $ take 100000 $ randomRs (0,(4.0e-45::Float)) $ mkStdGen 0)
```

... still runs imperceptably fast :)

10 just seems a bit low for this sort of test, not that I can say objectively why 100000 is necessarily better.

File: test/Spec.hs [hunk 2]
```diff
@@ -71,8 +72,21 @@ main =
...
     -- , bitmaskSpec @Word
     , runSpec
+    , unitTests
     ]
 
+unitTests :: TestTree
+unitTests = testGroup "Unit tests"
+  [ -- Check that https://github.com/haskell/random/issues/53 does not regress
+
+    testCase "Subnormal generation respects upper bound" $
```
Comment (curiousleo) [inline test/Spec.hs]: The other specs have the type in their title. My suggestion would be to have a `testGroup "(Float)"` and put the `testCase`s in there.

"respects" is a bit ambiguous here, perhaps "does not exceed" is clearer?

**Review by curiousleo — COMMENTED** [review]
It is not clear to me how we avoid these issues, given that we're not doing anything sophisticated with floating point ranges.

Still, this is definitely good to have. I've added a few comments.

**Comment by idontgetoutmuch** [comment]
I am going to merge as it just beefs up testing.

**Pull request merged**